service/src/main/java/com/laiteam/echowall/service/imp/VerificationCodeServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.laiteam.echowall.service.imp;
+
+import com.laiteam.echowall.dal.entity.User;
+import com.laiteam.echowall.dal.entity.UserVerificationCode;
+import com.laiteam.echowall.dal.repository.UserRepository;
+import com.laiteam.echowall.dal.repository.UserVerificationCodeRepository;
+import com.laiteam.echowall.service.VerificationCodeService;
+import com.laiteam.echowall.service.util.OptionalUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Example;
+import org.springframework.data.domain.ExampleMatcher;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+
+@Service
+public class VerificationCodeServiceImpl implements VerificationCodeService {
+
+    final UserVerificationCodeRepository userVerificationCodeRepository;
+
+    @Autowired
+    public VerificationCodeServiceImpl(UserVerificationCodeRepository userVerificationCodeRepository){
+        this.userVerificationCodeRepository = userVerificationCodeRepository;
+    }
+
+    @Override
+    public String generateVerificationCode(int length) {
+        Random generator = new Random();
```

Comment:
verification code should be fixed length, shouldn't use dynamic length
1. rename generateSixDigitsVerificationCode
2. move generateNDigits(int length) to a helper function